### ISSUE TITLE
Inconsistent behaviour when converting tuple to list

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#5553]
**Describe the bug**

First case type checked fine, but second has error:

```python
choices = (
    ("a", "b"),
    ("c", "d"),
)


def items() -> list[tuple[str, str]]:
    return list(choices)


def items_bad() -> list[tuple[str, str]]:
    converted = list(choices)
    return converted # -- error here
```

```
error: Expression of type "list[tuple[Literal['a'], str] | tuple[Literal['b'], str]]" cannot be assigned to return type "list[tuple[str, str]]"
    "list[tuple[Literal['a'], str] | tuple[Literal['b'], str]]" is incompatible with "list[tuple[str, str]]"
      Type parameter "_T@list" is invariant, but "tuple[Literal['a'], str] | tuple[Literal['b'], str]" is not the same as "tuple[str, str]" (reportGeneralTypeIssues)
```
**To Reproduce**
See code.

**Expected behavior**
This code should produce no errors.

**VS Code extension or command-line**
pyright 1.1.318

**Additional context**
mypy works fine in this case.

**Comment by last-partizan** [comment]
it looks like duplicate of https://github.com/microsoft/pyright/issues/5485

But then there's another bug, why first case does not produces error?

**Comment by erictraut** [comment]
This behavior is correct. The `items` function uses [bidirectional type inference](https://microsoft.github.io/pyright/#/type-inference?id=bidirectional-type-inference-expected-types) to inform the evaluated type. The `items_bad` method uses normal evaluation rules. If you to eliminate type error, you need to provide an explicit type annotation for `converted`.

```python
def items_bad() -> list[tuple[str, str]]:
    converted: list[tuple[str, str]] = list(choices)
    return converted
```
